modules/bibupload/lib/bibupload.py

```diff
@@ -674,7 +679,35 @@ def find_record_from_oaiid(oaiid):
         write_message("   Error during find_record_from_oaiid(): %s " % error,
                       verbose=1, stream=sys.stderr)
     if res:
-        return res[0][0]
+        if len(res) == 1:
+            return res[0][0]
+        else:
+            # Principle dupes detected.
+            # If the dupes are resolved only one record should have survived,
+            # and all others are deleted. => get non-deleted record(s)
+            lst = []
+            for r in res:
+                lst.append('recid:"%s"' % str(r[0]))
+            candidates = search_pattern(p=' or '.join(lst))
+            deleted    = search_pattern(p='980__c:"DELETED"')
```

Comment:
>> +            deleted    = search_pattern(p='980__c:"DELETED"')
> 
> It's perfect 980:"DELETED"

This and a minor fix for handling the return of None is in the updated PR. (BTW: shouln't the intbitset.add(None) just work?)
 
-- 
Kind regards,

Alexander Wagner

Deutsches Elektronen-Synchrotron DESY
Library and Documentation

Building 01d Room OG1.444
Notkestr. 85
22607 Hamburg

phone:  +49-40-8998-1758
fax:    +49-40-8994-1758
e-mail: alexander.wagner@desy.de